onnxruntime/core/providers/cuda/tensor/transpose_impl.cu

```diff
@@ -7,44 +7,52 @@
 namespace onnxruntime {
 namespace cuda {
 
-constexpr unsigned int TILE_DIM = 16;
+constexpr unsigned int NUM_ELE_PER_THREAD = 4;
```

Comment:
Have we ever tried other values like 2/8/16?

Reply:
And we expect same improvement on ROCM?

Reply:
Tried 4 and 8, make no much difference. Most of our other kernels use 4, so follow the same number here. Didn't check the perf on ROCm.